diff --git a/ios/RNWatch/RNWatch.h b/ios/RNWatch/RNWatch.h
index abb8fa56bc21d82aac74858efb0979c671b3e2f2..535cd8f3eecc6b3bd09916ffa6e3eaf3fd5f9141 100644
--- a/ios/RNWatch/RNWatch.h
+++ b/ios/RNWatch/RNWatch.h
@@ -1,20 +1,5 @@
-// import RCTBridgeModule.h
-#if __has_include("RCTBridgeModule.h")
-#import "RCTBridgeModule.h"
-#elif __has_include(<React/RCTBridgeModule.h>)
-#import <React/RCTBridgeModule.h>
-#else
 #import "React/RCTBridgeModule.h"
-#endif
-
-// import RCTEventEmitter.h
-#if __has_include("RCTEventEmitter.h")
-#import "RCTEventEmitter.h"
-#elif __has_include(<React/RCTEventEmitter.h>)
-#import <React/RCTEventEmitter.h>
-#else
 #import "React/RCTEventEmitter.h"
-#endif
 
 @import WatchConnectivity;
 
