diff --git a/android/src/main/java/com/reactnativeshortcuts/ShortcutsModule.kt b/android/src/main/java/com/reactnativeshortcuts/ShortcutsModule.kt
index ec847cc11ec842a94a9689b9cbfd313fcb3db599..e12d52371d972a7504893896c2d14d08a436ef9f 100644
--- a/android/src/main/java/com/reactnativeshortcuts/ShortcutsModule.kt
+++ b/android/src/main/java/com/reactnativeshortcuts/ShortcutsModule.kt
@@ -33,11 +33,11 @@ class ShortcutsModule(reactContext: ReactApplicationContext) :
         super.onCatalystInstanceDestroy()
     }
 
-    override fun onActivityResult(activity: Activity?, requestCode: Int, resultCode: Int, data: Intent?) {
+    override fun onActivityResult(activity: Activity, requestCode: Int, resultCode: Int, data: Intent?) {
         // No implementation needed
     }
 
-    override fun onNewIntent(intent: Intent?) {
+    override fun onNewIntent(intent: Intent) {
         emitEvent(intent)
     }
 
@@ -53,7 +53,7 @@ class ShortcutsModule(reactContext: ReactApplicationContext) :
         }

         val context = reactApplicationContext ?: return
-        val activity = currentActivity ?: return
+        val activity = reactApplicationContext.currentActivity ?: return

         val shortcutItems = items.toArrayList().mapIndexed { index, _ ->
             val map = items.getMap(index) ?: return
@@ -94,7 +94,7 @@ class ShortcutsModule(reactContext: ReactApplicationContext) :
             promise.reject(NotSupportedException)
         }

-        val shortcutManager = currentActivity?.getSystemService<ShortcutManager>(ShortcutManager::class.java)
+        val shortcutManager = reactApplicationContext.currentActivity?.getSystemService<ShortcutManager>(ShortcutManager::class.java)
         val shortcutItems = shortcutManager?.dynamicShortcuts?.map {
             ShortcutItem(it.id, it.longLabel.toString(), it.shortLabel.toString(), null, null)
         }
@@ -109,7 +109,7 @@ class ShortcutsModule(reactContext: ReactApplicationContext) :
             promise.reject(NotSupportedException)
         }

-        val shortcutItem = getShortcutItemFromIntent(currentActivity?.intent)
+        val shortcutItem = getShortcutItemFromIntent(reactApplicationContext.currentActivity?.intent)

         promise.resolve(shortcutItem?.toMap())
     }
@@ -121,7 +121,7 @@ class ShortcutsModule(reactContext: ReactApplicationContext) :
             return
         }

-        val shortcutManager = currentActivity?.getSystemService<ShortcutManager>(ShortcutManager::class.java)
+        val shortcutManager = reactApplicationContext.currentActivity?.getSystemService<ShortcutManager>(ShortcutManager::class.java)
         shortcutManager?.removeAllDynamicShortcuts()
     }

@@ -126,7 +126,7 @@ class ShortcutsModule(reactContext: ReactApplicationContext) :
     }
 
     private fun getShortcutItemFromIntent(intent: Intent?): ShortcutItem? {
-        if (intent?.action !== INTENT_ACTION_SHORTCUT) {
+        if (intent?.action != INTENT_ACTION_SHORTCUT) {
             return null
         }
 
