diff --git a/src/DateTimePickerModal.ios.js b/src/DateTimePickerModal.ios.js
index 6ef0b0ac0d0e68a6fa8be1084a074cc28e3cfc76..122ea01319edde84356a5d5e071529664e080e61 100644
--- a/src/DateTimePickerModal.ios.js
+++ b/src/DateTimePickerModal.ios.js
@@ -6,6 +6,7 @@ import {
   TouchableHighlight,
   View,
   Appearance,
+  Platform,
 } from "react-native";
 import DateTimePicker from "@react-native-community/datetimepicker";
 import Modal from "./Modal";
@@ -15,7 +16,8 @@ export const BACKGROUND_COLOR_LIGHT = "white";
 export const BACKGROUND_COLOR_DARK = "#0E0E0E";
 export const BORDER_COLOR = "#d5d5d5";
 export const BORDER_COLOR_DARK = "#272729";
-export const BORDER_RADIUS = 13;
+export const BORDER_TOP_RADIUS = Platform.Version >= 26 ? 30 : 13;
+export const BORDER_BOTTOM_RADIUS = Platform.Version >= 26 ? 50 : 13;
 export const BUTTON_FONT_WEIGHT = "normal";
 export const BUTTON_FONT_COLOR = "#007ff9";
 export const BUTTON_FONT_SIZE = 20;
@@ -178,7 +180,7 @@ export class DateTimePickerModal extends React.PureComponent {
               display={display || "spinner"}
               {...otherProps}
               value={this.state.currentDate}
-              onChange={this.handleChange}
+              onValueChange={this.handleChange}
               // Recent versions @react-native-community/datetimepicker (at least starting with 6.7.0)
               // have a peculiar iOS behaviour where sometimes, for example in react-native Modal,
               // the inline picker is not rendered correctly if in datetime mode. Explicitly setting the height
@@ -224,8 +226,11 @@ const pickerStyles = StyleSheet.create({
     marginBottom: isIphoneX() ? 34 : 10,
   },
   container: {
-    borderRadius: BORDER_RADIUS,
-    marginBottom: 8,
+    borderTopLeftRadius: BORDER_TOP_RADIUS,
+    borderTopRightRadius: BORDER_TOP_RADIUS,
+    borderBottomLeftRadius: BORDER_BOTTOM_RADIUS,
+    borderBottomRightRadius: BORDER_BOTTOM_RADIUS,
+    marginBottom: Platform.Version >= 26 ? 2 : 8,
     overflow: "hidden",
   },
   pickerSpinner: {
@@ -340,7 +345,7 @@ export const CancelButton = ({
 
 export const cancelButtonStyles = StyleSheet.create({
   button: {
-    borderRadius: BORDER_RADIUS,
+    borderRadius: BORDER_BOTTOM_RADIUS,
     height: 57,
     justifyContent: "center",
   },
