diff --git a/node_modules/react-native-date-picker/ios/RNDatePickerManager.mm b/node_modules/react-native-date-picker/ios/RNDatePickerManager.mm
index e5e331b..2a535c8 100644
--- a/node_modules/react-native-date-picker/ios/RNDatePickerManager.mm
+++ b/node_modules/react-native-date-picker/ios/RNDatePickerManager.mm
@@ -51,7 +51,7 @@ RCT_CUSTOM_VIEW_PROPERTY(date, id, RNDatePicker)
 {
     NSDate *date = [self convertToNSDate:json];
     if (date) {
-        [(RNDatePicker *)view setDate:date];
+        [(DatePicker *)view setDate:date];
     }
 }
 
@@ -61,7 +61,7 @@ RCT_CUSTOM_VIEW_PROPERTY(minimumDate, id, RNDatePicker)
 {
     NSDate *date = [self convertToNSDate:json];
     if (date) {
-        [(RNDatePicker *)view setMinimumDate:date];
+        [(DatePicker *)view setMinimumDate:date];
     }
 }
 
@@ -69,7 +69,7 @@ RCT_CUSTOM_VIEW_PROPERTY(maximumDate, id, RNDatePicker)
 {
     NSDate *date = [self convertToNSDate:json];
     if (date) {
-        [(RNDatePicker *)view setMaximumDate:date];
+        [(DatePicker *)view setMaximumDate:date];
     }
 }
 
