diff --git a/node_modules/react-native-watch-connectivity/ios/RNWatch/RNWatch.h b/node_modules/react-native-watch-connectivity/ios/RNWatch/RNWatch.h
index abb8fa5..535cd8f 100644
--- a/node_modules/react-native-watch-connectivity/ios/RNWatch/RNWatch.h
+++ b/node_modules/react-native-watch-connectivity/ios/RNWatch/RNWatch.h
@@ -1,20 +1,5 @@
-// import RCTBridgeModule.h
-#if __has_include("RCTBridgeModule.h")
-#import "RCTBridgeModule.h"
-#elif __has_include(<React/RCTBridgeModule.h>)
-#import <React/RCTBridgeModule.h>
-#else
 #import "React/RCTBridgeModule.h"
-#endif
-
-// import RCTEventEmitter.h
-#if __has_include("RCTEventEmitter.h")
-#import "RCTEventEmitter.h"
-#elif __has_include(<React/RCTEventEmitter.h>)
-#import <React/RCTEventEmitter.h>
-#else
 #import "React/RCTEventEmitter.h"
-#endif
 
 @import WatchConnectivity;
 
